src/main/java/bio/overture/songsearch/graphql/AnalysisDataFetcher.java

```diff
@@ -56,4 +68,74 @@ public AnalysisDataFetcher(AnalysisService analysisService) {
       return analysisService.getAnalyses(filter.build(), page.build());
     };
   }
+
+  public DataFetcher<List<SampleMatchedAnalysisPair>> getSampleMatchedAnalysisPairsFetcher() {
+    return env -> {
+      val analysisId = env.getArguments().get("analysisId").toString();
+
+      val analysisFromId = analysisService.getAnalysisById(analysisId);
+      val samples = getFlattenedSamplesFromAnalysis(analysisFromId);
+
+      validateAnalysisAndSamplesValidForQuery(analysisFromId, samples);
```

Comment:
:thinking:  hmm  I guess what I'm trying to do here is kinda like a precondition for this query. But you raise a good point, the fact that I have to do it this way is probably a indication that this is not the right approach. Maybe the proper thing to do would be to short circuit and return empty data because that analysisId can't be matched..